File: cos/olympus/src/main/java/cos/olympus/game/events/Death.java
package cos.olympus.game.events;

import cos.olympus.game.Creature;

public record Death(
        int id,
        int tick,
        Spell spell,
        Creature victim
) implements Event {

    public cos.ops.out.Death toUserOp(int userId) {
        return new cos.ops.out.Death(id, tick, userId, spell.source().id(), victim.id());
    }

    @Override
    public String toString() {
        return "Death{" +
                "id=" + id +
                ", tick=" + tick +
                ", victim=" + victim.id() +
                ", spell=" + spell.id() +
                '}';
    }
}

File: cos/src/main/java/cos/olympus/game/events/Death.kt
package cos.olympus.game.events

import cos.olympus.game.Actor
import cos.ops.out.Death

data class Death(
    override val id: Int,
    override val tick: Int,
    val spell: Spell,
    val victim: Actor
) : Event {
    fun toUserOp(userId: Int) = Death(id, userId, spell.source.id, victim.id)
}

